app/meteo/meteo.component.ts

```diff
@@ -12,23 +12,21 @@ export class MeteoComponent {
     @Input() position: CurrentPosition;
     @Input() updated: City[];
 
+    storeNames: City[];
+    weatherStoreValue: City[];
+
     constructor(private weatherService: WeatherService) {
     }
 
-    onAdd(value: City[]): void {
+    onAdd(value: string): void {
         let newValue: City[] = this.weatherService.getStore().slice(0);
```

Comment:
I use newValue for changing reference to storage. Hence ngOnChanges event triggers every time, when I add city to table and update storage array.